src/com/hms/flexyosisoftconnector/Tag.java

```diff
@@ -120,6 +193,417 @@ public void recordTagValue(String time)
+    *
+    * The tagTimeIntFileName file holds a time tracker for every tag. One file exists per tag.
+    * This file will hold a string which will be the epoch time in seconds for where/when
+    * in the Flexy historical logs has the data queue left off on sending data to OSIsoft
+    *
+    * The file can only contain numbers because it is converted to a long
+    *
+    * The below function makes sure the file exists, if not add in the current time to that
+    * otherwise read off the epoch time in that file
+    *
+    * @return Returns a string of the file contents. it will be the epoch time tracker for the data queue.
+    * The value is called a timeInt in the return file from the EBD commands, so the name will be timeInt throughout code.
+    */
+   public String readTimeTrackerFile()
+   {
+      String timeInt = TIME_ERROR;
+      // check if file exists
+      File tempFile = new File(tagTimeIntFileName);
+      boolean timeTrackerFileExists = tempFile.exists();
+
+      if(timeTrackerFileExists) {
+         BufferedReader reader;
+         try {
+            reader = new BufferedReader(new FileReader(tagTimeIntFileName));
+            String line = reader.readLine();
+            while (line != null) {
+
+               // There will be just one line, but just save the last line.
+               timeInt = line;
+               line = reader.readLine();
+            }
+            reader.close();
+         } catch (IOException e) {
+             e.printStackTrace();
+             Logger.LOG_ERR("Something went wrong reading tag data queue time tracker from timeInt file. Check tag timeInt file: "+tagTimeIntFileName);
+         }
+
+      } else { // When the connector has not run before, these files will not exist.
+
+          // If file does not exist, set timeInt to the current time to start a new queue
+          Logger.LOG_INFO(getTagName() + ": time int was null, setting back of tag data queue current time");
+          long tmpTimeIntMilliSeconds = System.currentTimeMillis();
+          // Current time (when the list was null) will be the new back of queue
+          String writtenValue = writeTimeTrackerFile(tmpTimeIntMilliSeconds);
+          // Return what got written in the writeTimeTrackerFile function
+          return writtenValue;
+      }
+
+      // Return the string read in this function
+      return timeInt;
+   }
+
+    /**
+     * @param x Decimal value to transform to binary
+     * @param len How many 1's and 0's should we have?
+     * This adds in the missing leading 0's to the generated binary string
+     * when given small inputs.
+     *
+     * Example, x input of 4 would be "100"
+     * x output with len input of 8 would be "00000100"
+     *
+     * @return Returns the string of the decimal value converted to binary
+     * with leading 0's added if needed.
+     */
+    public static String toBinary(int x, int len) {
+        StringBuilderLite result = new StringBuilderLite(10);

```

Comment:
Starting to loose my ability to concentrate. Ill stop reviewing for now, 10 seems pretty magical here and I cant figure out the reason for it. Seems like if len > 10 this would all break?

Reply:
uh, that should be len instead of 10.  It should also be 8 since len will always be 8... I don't know why i left that there.